src/cluster.c

```diff
@@ -1430,14 +1430,49 @@ void clusterCommandSlots(client * c) {
         /* Add cluster slots info when occur different node with start
          * or end of slot. */
         if (i == CLUSTER_SLOTS || n != getNodeBySlot(i)) {
-            addNodeReplyForClusterSlot(c, n, start, i-1);
+            addNodeReplyForClusterSlot(recording_client, n, start, i-1);
             num_masters++;
             if (i == CLUSTER_SLOTS) break;
             n = getNodeBySlot(i);
             start = i;
         }
     }
-    setDeferredArrayLen(c, slot_replylen, num_masters);
+    setDeferredArrayLen(recording_client, slot_replylen, num_masters);
+    return stopCaching(recording_client);
+}
+
+int verifyCachedClusterSlotsResponse(sds cached_response) {
+    sds generated_response = generateClusterSlotResponse();
+    int result = !sdscmp(generated_response, cached_response);
+    if (!result) serverLog(LL_WARNING,"\ngenerated_response:\n%s\n\ncached_response:\n%s", generated_response, cached_response);
```

Comment:
lower to NOTICE? since there isn't much an admin can do about this? Also, the strings could be made more descriptive IMO because there won't be much context in the log as in the code here.

```suggestion
    if (!result) serverLog(LL_NOTICE,"\nstale CLUSTER SLOTS response cache found; expected: %s\n\ngot:\n%s", generated_response, cached_response);
```

Reply:
This is only printed when the system is about to crash, so we bumped this to LL_WARNING to make sure it was written out if debug assertions were enabled.

Reply:
Make sense. I missed the crashing part

Reply:
Might be worth documenting then, since a bunch of people commented about this.